### DISCUSSION TITLE
Reducing the number of partitions

### DISCUSSION THREAD

**Discussion by FestplattenSchnitzel** [akvorado/akvorado#1508] [Q&A]
In the past, I've had Akvorado configured with
* `clickhouse.kafka.consumers: 12`
* `kafka.topic-configuration.num-partitions: 12`

for testing purposes, but changed both back to `5` some time ago.

However, running `kafka-consumer-groups.sh --bootstrap-server localhost:9092 --describe --group clickhouse` still shows 12 "active" partitions (ones with values at `CURRENT-OFFSET`, `LOG-END-OFFSET` and `LAG`) for one topic flows-243Q[…] and, surprising to me, six ones for another topic flows-LAAB[…] where no data seems to go through (no values for `CURRENT-OFFSET`, `LOG-END-OFFSET` and `LAG`) but still for each partition there is a clickhouse consumer and client connected.  Any idea what's up with these?

Regarding the number of partitions, there's also this log entry:
```
Nov 18 10:47:57 akvorado akvorado[872591]: {"level":"warn","brokers":"127.0.0.1:9092","topic":"flows-243QR3DRGFU3LAWDAIZA7TXBYM","time":"2024-11-18T10:47:57+01:00","caller":"akvorado/orchestrator/kafka/root.go:93","module":"akvorado/orchestrator/kafka","message":"cannot decrease the number of partitions (from 12 to 5)"}
```

OK, a quick search reveals: [reducing the number of partitions isn't supported by Kafka](https://stackoverflow.com/questions/45497878/how-to-decrease-number-partitions-kafka-topic).

Is there some way to reduce the number of partitions (and more important the number of consumers) anyway?  I wouldn't care about data loss as I have huge performance problems currently (12 clickhouse consumers on 12 CPU cores don't work out very well).

I've already tried deleting all of `/var/lib/kafka` but it didn't work.

That's in advance.

**Comment by FestplattenSchnitzel** [comment]
Ah, just noticed, the number of consumers is already down to five, but of course one consumer now consumes from multiple topics.

**Reply by vincentbernat** [reply]
You can have several topics after upgrades or when changing the schema. After some time, only one topic is active. Only the active topic is being configured.
